var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -241,20 +267,16 @@ def cmake_args(self):
             'GMX_USE_RDTSCP', str(target.family) in ('x86_64', 'x86')
         ))
 
-        if '+mdrun_only' in self.spec:
-            options.append('-DGMX_BUILD_MDRUN_ONLY:BOOL=ON')
-        else:
-            options.append('-DGMX_BUILD_MDRUN_ONLY:BOOL=OFF')
+        if self.spec.satisfies('@:2020'):
+            options.append(self.define_from_variant('GMX_BUILD_MDRUN_ONLY', 'mdrun_only'))
 
         if '~openmp' in self.spec:
             options.append('-DGMX_OPENMP:BOOL=OFF')
         else:
             options.append('-DGMX_OPENMP:BOOL=ON')
```

Comment:
```suggestion
        options.append(self.define_from_variant('GMX_OPENMP', 'openmp'))
```

If either `~openmp` or `+openmp` will always be in `self.spec`, then it is appropriate to set both cases explicitly. The GROMACS CMake logic is unconventional here, and there doesn't seem to be any other appropriate way for Spack to account for that.